src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcReportTechnicalIssueHelperSpec.scala

```diff
@@ -0,0 +1,174 @@
+
+import org.jsoup.Jsoup
+import org.scalatest.{Matchers, WordSpecLike}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.i18n.Lang
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.hmrcfrontend.MessagesSupport
+import uk.gov.hmrc.hmrcfrontend.views.html.helpers._
+
+class hmrcReportTechnicalIssueHelperSpec
+    extends WordSpecLike
+    with Matchers
+    with MessagesSupport
+    with GuiceOneAppPerSuite {
+
+//  val notPlay25 = sys.env("PLAY_VERSION") != "2.5"
+
+  override def fakeApplication(): Application =
+    new GuiceApplicationBuilder()
+      .configure(
+        Map(
+          "contact-frontend.serviceCode" -> "my-service",
+          "platform.frontend.host"       -> "https://www.tax.service.gov.uk"
+        ))
+      .build()
+
+  def buildApp(properties: Map[String, String] = Map.empty): Application =
+    new GuiceApplicationBuilder()
+      .configure(properties)
+      .build()
+
+  "HmrcReportTechnicalIssueHelper" should {
+    "render link when serviceCode is set" in {
+      implicit val fakeRequest = FakeRequest("GET", "/foo")
+      implicit val app         = buildApp(Map("contact-frontend.serviceCode" -> "my-service"))
+
+      val content = contentAsString(HmrcReportTechnicalIssueHelper()(messages, fakeRequest))
+      val links   = Jsoup.parse(content).select("a")
+
+      links                    should have size 1
+      links.first.attr("href") shouldBe "https://www.tax.service.gov.uk/contact/problem_reports_nonjs?newTab=true&service=my-service&referrerUrl=https%3A%2F%2Fwww.tax.service.gov.uk%2Ffoo"
+    }
+    "render link with correct referrer URL" in {
+      implicit val fakeRequest = FakeRequest("GET", "/")
+      implicit val app         = buildApp(Map("contact-frontend.serviceCode" -> "my-service"))
+
+      val content = contentAsString(HmrcReportTechnicalIssueHelper()(messages, fakeRequest))
+      val links   = Jsoup.parse(content).select("a")
+
+      links                    should have size 1
+      links.first.attr("href") shouldBe "https://www.tax.service.gov.uk/contact/problem_reports_nonjs?newTab=true&service=my-service&referrerUrl=https%3A%2F%2Fwww.tax.service.gov.uk%2F"
+    }
+    "render link with query string in referrer URL" in {
+      implicit val fakeRequest = FakeRequest("GET", "/abc/def?ghi=jkl&mno=pqr")
+      implicit val app         = buildApp(Map("contact-frontend.serviceCode" -> "my-service"))
+
+      val content = contentAsString(HmrcReportTechnicalIssueHelper()(messages, fakeRequest))
+      val links   = Jsoup.parse(content).select("a")
+
+      links                    should have size 1
+      links.first.attr("href") shouldBe "https://www.tax.service.gov.uk/contact/problem_reports_nonjs?newTab=true&service=my-service&referrerUrl=https%3A%2F%2Fwww.tax.service.gov.uk%2Fabc%2Fdef%3Fghi%3Djkl%26mno%3Dpqr"
+    }
+//    if (notPlay25) {
```

Comment:
Could we move this to a Play-26 specific folder to avoid having commented out code?

Reply:
I couldn't get that working - but that was in the middle of a few other things I was trying to sort out.  I'll try again now.

Reply:
It's working now, thanks.